compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/SymbolTable.java

```diff
@@ -626,7 +626,7 @@ private void defineIntegerUnaryOperations() {
         BType[] intTypes = {intType, byteType, signed32IntType, signed16IntType, signed8IntType, unsigned32IntType,
                 unsigned16IntType, unsigned8IntType};
         for (BType type : intTypes) {
-            defineUnaryOperator(OperatorKind.ADD, type, intType);
+            defineUnaryOperator(OperatorKind.ADD, type, type);
```

Comment:
Still doesn't work for user-defined subtypes.

```ballerina
type Foo -1|1;

public function main() {
    Foo g = -1;
    Foo _ = +g; // error: incompatible types: expected 'Foo', found 'int'
}
```

Reply:
Will fix this